### PR TITLE
2020 17 03 api gateways

### PR DESCRIPTION
**Pull request by sebbsan**
post of my API Gateway talk

### CODE REVIEW

File: content/posts/2020-03-19-api-gateways.md [hunk 1]
```diff
@@ -257,16 +257,16 @@ Yes it‘s a Swiss Army Knife, but more of the small, practical sort. The declar
...
 
 ### Apollo Gateway
 
 [Apollo Gateway](https://www.apollographql.com/docs/apollo-server/api/apollo-gateway/) is a GraphQL gateway to federate other GraphQL APIs
 
 #### The Good
 
-- More a Framework kind of thing
-- Made by early GraphQL Adopters
+- More a Framework kind of thing, designed to build your own custom gateway
```
Comment (george-haddad) [inline content/posts/2020-03-19-api-gateways.md]: This sounds odd, I think you meant to say "More of a framework kind of thing,"

File: content/posts/2020-03-19-api-gateways.md [hunk 2]
```diff
@@ -275,7 +275,7 @@ Powerful reverse proxy with added functionality– worth trying if you run K8S.
 
 #### Conclusion
 
-Only for nishe usecase when you want / need to federate other GraphQL APIs.
+Only for niche usecase when you want / need to federate other GraphQL APIs.
```
Comment (george-haddad) [inline content/posts/2020-03-19-api-gateways.md]: Suggestion to rephrase "Only for niche use cases where federating other GraphQL APIs is needed"

File: content/posts/2020-03-19-api-gateways.md [hunk 3]
```diff
@@ -305,4 +305,4 @@ You can find the talk presentation here:
 
 ## The End
 
-That concludes my API Gateway article. Please note, a lot of things written above are my personal opinions. If you have questions or comments, feel free to get in touch with me.
+That concludes my API Gateway overview. Please note, a lot of things written above are my personal opinions. If you have questions or comments, feel free to get in touch with me.
```
Comment (george-haddad) [inline content/posts/2020-03-19-api-gateways.md]: Rephrase suggestion  "Please note that these were my personal opinions and should there be any questions or comments, then please feel free to get in touch with me"

File: content/posts/2020-03-19-api-gateways.md [hunk 4]
```diff
@@ -22,19 +22,281 @@ It all started with an architecture pattern that turned into really powerful too
...
+- Provide an API Portal
+- The silver bullet for all your problems (Quote: Vendor XYZ)
 
-API Gateways as an architecture pattern was canonized with the arrival of microservices architecture, but it was a well known technique long before that.
+## Common Patterns
 
-## API Gateway - Alternatives
+### The "Olden Days"
+
+API Gateways as an architecture pattern was canonized with the arrival of microservices architecture, but it was a well known technique long before that. With the arrival of mobile and Ajax, as well as REST becoming fashionable, an API Gateway was often built a separate Application that was forwarding API Calls from the outside. Maybe you mapped to a legacy SOAP API or to another legacy stack.
```
Comment (george-haddad) [inline content/posts/2020-03-19-api-gateways.md]: ... often built **as** a separate application ...

No need to capitalize the word application in the middle of a sentence, same thing goes for the word "Calls" afterwards.

File: content/posts/2020-03-19-api-gateways.md [hunk 5]
```diff
@@ -22,19 +22,281 @@ It all started with an architecture pattern that turned into really powerful too
...
+
+![API Gateway The History](/apigateway/api-gateway-scenarios-Page-1.png "API Gateways - The History")
+
+### The RPC or CQRS Gateway
+
+Let's say our backend doesn’t follow REST Architecture. We still need to provide an API for outside clients to connect. We can use asynchronous designs in the backend such as RPC or CQRS using protocols such as JMS, ActiveMQ, RabbitMQ, or Protocol Buffers, or we have a "Service Bus", and messages need go into the bus from the outside. In this case, putting an API Service, aka. API Gateway, in front will do the job, and it will also give us the flexibility to add REST-style Resources. Some products we will discuss later come with adapters for common protocols such as for Kafka, AMQP and more.
```
Comment (george-haddad) [inline content/posts/2020-03-19-api-gateways.md]: Remove the  `, or ` before Protocol Buffers.  "Resources" does not need to be capitalized.

File: content/posts/2020-03-19-api-gateways.md [hunk 6]
```diff
@@ -0,0 +1,335 @@
...
+
+All of those companies come with their own set of challenges, be it how to hit the market as fast as possible with a high quality product, or operate and run a financial platform with extremely high levels of availability and inpenetrable security.
+
+If you want to be part of this, you can find open tech positions across the whole portfolio here: <https://www.finleap.com/careers/tech/>
+
+It's a great opportunity to learn from each other, understand how these challenges were mastered in the past, and share a beer and a laugh together.
+
+At the last meetup in March, I gave a talk about **API Gateways** - a common pattern about what they are, where they came from, and a short and totally subjective overview of the available solutions on the market. I came across this pattern and technology in 2014, and was interested in it ever since, following the different development and products that hit the market.
```
Comment (ajantsch) [inline content/posts/2020-03-19-api-gateways.md]: I would remove `a common pattern` here or rephrase the sentence, like "... I gave a talk about the common pattern of API Gateways, what they are, where they came from..."

**Pull request merged**